libs/barista-components/experimental/datepicker/src/datepicker.ts

```diff
@@ -0,0 +1,459 @@
+export class DtDatepickerBase {
+  constructor(
+    public _parentForm: NgForm,
+    public _parentFormGroup: FormGroupDirective,
+    public ngControl: NgControl,
+  ) {}
+}
+export const _DtDatepickerBase = mixinTabIndex(mixinDisabled(DtDatepickerBase));
+
+@Component({
+  selector: 'dt-datepicker',
+  templateUrl: 'datepicker.html',
+  styleUrls: ['datepicker.scss'],
+  host: {
+    class: 'dt-datepicker',
+    '[attr.id]': 'id',
+    '[attr.aria-disabled]': 'disabled.toString()',
+  },
+  inputs: ['disabled', 'tabIndex'],
+  encapsulation: ViewEncapsulation.Emulated,
+  preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger('transformPanel', [
+      state(
+        'void',
+        style({
+          transform: 'scaleY(0) translateX(-1px)',
+          opacity: 0,
+        }),
+      ),
+      state(
+        'showing',
+        style({
+          opacity: 1,
+          transform: 'scaleY(1) translateX(-1px)',
+        }),
+      ),
+      transition(
+        'void => *',
+        group([
+          query('@fadeInContent', animateChild()),
+          animate('150ms cubic-bezier(0.25, 0.8, 0.25, 1)'),
+        ]),
+      ),
+      transition('* => void', [
+        animate('250ms 100ms linear', style({ opacity: 0 })),
+      ]),
+    ]),
+    trigger('fadeInContent', [
+      state('showing', style({ opacity: 1 })),
+      transition('void => showing', [
+        style({ opacity: 0 }),
+        animate('150ms 100ms cubic-bezier(0.55, 0, 0.55, 0.2)'),
+      ]),
+    ]),
+  ],
+})
+export class DtDatePicker<D>
+  extends _DtDatepickerBase
+  implements ControlValueAccessor, CanDisable, HasTabIndex, OnDestroy {
+  /** Unique id of the element. */
+  @Input()
+  get id(): string {
+    return this._id;
+  }
+  set id(value: string) {
+    this._id = value || this._uid;
+  }
+  private _id: string;
+  private _uid = `dt-datepicker-${uniqueId++}`;
+
+  /** Value of the datepicker control. */
+  @Input()
+  get value(): D | null {
+    return this._value;
+  }
+  set value(newValue: D | null) {
+    if (newValue !== this._value) {
+      this._value = newValue;
+      this._changeDetectorRef.markForCheck();
+    }
+  }
+  private _value: D | null = null;
+
+  /** The date to open the calendar to initially. */
+  @Input()
+  get startAt(): D | null {
+    return this._value || this._startAt;
+  }
+  set startAt(value: D | null) {
+    this._startAt = getValidDateOrNull(this._dateAdapter, value);
+  }
+  private _startAt: D | null;
+
+  /** Classes to be passed to the datepicker panel. Supports the same syntax as `ngClass`. */
+  // tslint:disable-next-line:no-any
+  @Input() panelClass: string | string[] | Set<string> | { [key: string]: any };
+
+  /** Property that enables the timepicker, so that a time can be entered as well. */
+  @Input() isTimeEnabled: boolean;
+
+  /** Property decides whether or not the today button should be shown. */
+  @Input() isTodayButtonShown = true;
```

Comment:
nit: Something like `showTodayButton` would sound more natural IMO.

Reply:
I'm also not sure about the default. Maybe we should default this to false, to make it an optIn rather than an optOut